### PR TITLE
WL-4200 Web Content links in template sites are not duplicated in new…

### PR DESCRIPTION
**Pull request by ouit0408**
… site.

Checking if the web content url has already been added in new site.if
clause was checking for webcontent urls which never were equal because of
the unique siteId for each site.introduced a constant to check for
equality .

### CODE REVIEW

File: web-impl/impl/src/java/org/sakaiproject/web/impl/WebServiceImpl.java [hunk 1]
```diff
@@ -422,10 +422,12 @@ public void transferCopyEntities(String fromContext, String toContext, List ids)
 							    for (ToolConfiguration config: toolConfs) {
 								if (config.getToolId().equals(TOOL_ID)) {
 								    SitePage p = config.getContainingPage();
+									// Checking if the 'source' property of fromSite and toSite are same by replacing respective siteIds with a constant.
+									String fromContentUrl = contentUrl.replace(fromContext, WEB_CONTENT_URL_PROP);
+									String toContentUrl = config.getPlacementConfig().getProperty(WEB_CONTENT_URL_PROP).replace(toContext,WEB_CONTENT_URL_PROP);
```
Comment (buckett) [inline web-impl/impl/src/java/org/sakaiproject/web/impl/WebServiceImpl.java]: Is there the possibility of a NPE here where there isn't a URL set on the tool?

Comment (ouit0408) [inline web-impl/impl/src/java/org/sakaiproject/web/impl/WebServiceImpl.java]: isn't this method called only for webcontent link type of page  which always will have a url?

Comment (buckett) [inline web-impl/impl/src/java/org/sakaiproject/web/impl/WebServiceImpl.java]: I mean the site it's copying into.

File: web-impl/impl/src/java/org/sakaiproject/web/impl/WebServiceImpl.java [hunk 2]
```diff
@@ -422,10 +422,13 @@ public void transferCopyEntities(String fromContext, String toContext, List ids)
 							    for (ToolConfiguration config: toolConfs) {
 								if (config.getToolId().equals(TOOL_ID)) {
 								    SitePage p = config.getContainingPage();
+									// Checking if the 'source' property of fromSite and toSite are same or not by replacing respective siteIds with a constant.
```
Comment (buckett) [inline web-impl/impl/src/java/org/sakaiproject/web/impl/WebServiceImpl.java]: Indentation looks wrong.

**Comment by ouit0408** [comment]
Have pushed the changes.

**Pull request merged**